content/post/115-protocol-labs-and-igalia-collaborating-on-dweb-in-browsers.md

```diff
@@ -0,0 +1,194 @@
+
+## Distributed web protocols in custom handlers
+
+<div style="width: 207pt; margin-left: auto; margin-right: auto;">
+<svg width="207pt" height="305pt" viewBox="0.00 0.00 207.04 305.00">
+<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 301)">
+<polygon fill="transparent" stroke="transparent" points="-4,4 -4,-301 203.04,-301 203.04,4 -4,4"/>
+<!-- 0 -->
+<g id="node1" class="node">
+<title>0</title>
+<polygon fill="none" stroke="black" points="103.04,-297 49.04,-297 49.04,-261 103.04,-261 103.04,-297"/>
+<text text-anchor="middle" x="76.04" y="-275.3" font-family="Times,serif" font-size="14.00">User</text>
+</g>
+<!-- 1 -->
+<g id="node2" class="node">
+<title>1</title>
+<polygon fill="none" stroke="black" points="114.54,-210 37.54,-210 37.54,-174 114.54,-174 114.54,-210"/>
+<text text-anchor="middle" x="76.04" y="-188.3" font-family="Times,serif" font-size="14.00">Browser</text>
+</g>
+<!-- 0&#45;&gt;1 -->
+<g id="edge1" class="edge">
+<title>0&#45;&gt;1</title>
+<path fill="none" stroke="black" d="M76.04,-260.8C76.04,-246.05 76.04,-224.92 76.04,-210.18"/>
+<text text-anchor="middle" x="111.54" y="-231.8" font-family="Times,serif" font-size="14.00">ifps://blah</text>
+</g>
+<!-- 2 -->
+<g id="node3" class="node">
+<title>2</title>
+<polygon fill="none" stroke="black" points="137.04,-123 15.04,-123 15.04,-87 137.04,-87 137.04,-123"/>
+<text text-anchor="middle" x="76.04" y="-101.3" font-family="Times,serif" font-size="14.00">HTTP gateway</text>
+</g>
+<!-- 1&#45;&gt;2 -->
+<g id="edge2" class="edge">
+<title>1&#45;&gt;2</title>
+<path fill="none" stroke="black" d="M76.04,-173.8C76.04,-159.05 76.04,-137.92 76.04,-123.18"/>
+<text text-anchor="middle" x="137.54" y="-144.8" font-family="Times,serif" font-size="14.00">http://ipfs.io/blah</text>
+</g>
+<!-- 3 -->
+<g id="node4" class="node">
+<title>3</title>
+<ellipse fill="none" stroke="black" cx="76.04" cy="-18" rx="76.09" ry="18"/>
+<text text-anchor="middle" x="76.04" y="-14.3" font-family="Times,serif" font-size="14.00">IPFS Network</text>
+</g>
+<!-- 2&#45;&gt;3 -->
+<g id="edge3" class="edge">
+<title>2&#45;&gt;3</title>
+<path fill="none" stroke="black" d="M76.04,-86.8C76.04,-72.05 76.04,-50.92 76.04,-36.18"/>
+<text text-anchor="middle" x="109.54" y="-57.8" font-family="Times,serif" font-size="14.00">/ipfs/blah</text>
+</g>
+</g>
+</svg>
+</div>
+
+An existing approach to use IPFS in browsers that don’t natively support this protocol is to rely on an [HTTP gateway](https://ipfs.github.io/public-gateway-checker/). Additionally, the redirection of IPFS links can be automatically performed using [HTML custom handlers](https://html.spec.whatwg.org/multipage/system-state.html#custom-handlers). However, this approach has several limitations:
```

Comment:
Can you link to the HTTP gateway docs? https://docs.ipfs.io/concepts/ipfs-gateway/